### PR TITLE
Assign players to trainings based on user selection.

### PR DESCRIPTION
**Pull request by juliomejia97**
Demo App.

https://github.com/LaZona16/zone_app/assets/19478156/a87daa4f-5a63-4e18-aae2-81330fd4aee7

This PR is intent to:

### Allowing selection based on the institution and category.
- [X] Call to backend
- [X] Mapping of models to be aligned with backend
- [X] Show a selectable list of players

### Feature to support the selection of a player list based on the training requirements.
- [X] If it is an individual training we don't show the player selected information.
image

- [X] If it is a competence training we show the information of each player and it's corresponding device.
image

Finally, with this PR we made a little bit of code refactoring.

### CODE REVIEW

File: lib/src/modules/training/training_statistics.dart [hunk 1]
```diff
@@ -11,14 +11,16 @@ class TrainingStatistics extends GetView<TrainingController> {
...
   Widget build(BuildContext context) {
-    return Column(
-      children: [
-        const BuildTime(),
-        const WallSettings(),
-        controller.trainingType == TrainingType.individual
-            ? const Statistics()
-            : const CompetitionStatistics(),
-      ],
+    return SingleChildScrollView(
```
Comment (crissebasbol) [inline lib/src/modules/training/training_statistics.dart:14]: 👍

File: lib/src/modules/players_list/controller/player_list_controller.dart [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    } else {
+      final Training training = arguments[trainingArgument];
+      if (training == Training.twoWalls && selectedPlayers.length != 2) {
+        Get.snackbar('Error de Selección', 'Debe seleccionar 2 jugadores');
+        return;
+      }
+
+      if (training == Training.threeWalls && selectedPlayers.length != 3) {
+        Get.snackbar('Error de Selección', 'Debe seleccionar 3 jugadores');
+        return;
```
[the remark this answers is not in the document]
Comment (juliomejia97) [reply lib/src/modules/players_list/controller/player_list_controller.dart:90]: The idea is that this data comes from the back-end at least

File: lib/src/core/strings.dart [hunk 3]
```diff
@@ -43,3 +43,6 @@ const String CATEGORY_LABEL = 'Categoría.';
 const String CATEGORY_PLACEHOLDER = 'Seleccione una categoría';
 const String REGISTER_BUTTON_TEXT = 'Registrar';
 const String DATEPICKER_LABEL = 'Fecha De Nacimiento.';
+const String AGE_LABEL = 'Edad.';
+const String ACCEPT_BUTTON_TEXT = 'Aceptar';
+const String SAVE_TITLE = 'SAVE';
```
[the remark this answers is not in the document]
Comment (juliomejia97) [reply lib/src/core/strings.dart:48]: No yet

File: lib/src/domain/data/repository/zone_api_url.dart [hunk 4]
```diff
@@ -4,4 +4,5 @@ class ZoneApiUrl {
   static const String getAllInstitutions = '$base/Institution/';
   // Players
   static const String createPlayer = '$base/Player';
+  static const String getPlayers = '$base/Player?';
```
[the remark this answers is not in the document]
Comment (juliomejia97) [reply lib/src/domain/data/repository/zone_api_url.dart:7]: Thank you!

File: lib/src/domain/model/hit_position.dart [hunk 5]
```diff
@@ -0,0 +1,11 @@
+enum HitPosition {
```
[the remark this answers is not in the document]
Comment (juliomejia97) [reply lib/src/domain/model/hit_position.dart:4]: Do you consider this should be done here or we can manage this on a future ticket

[the remark this answers is not in the document]
Comment (juliomejia97) [reply lib/src/domain/model/hit_position.dart:4]: Got it

File: lib/src/modules/players_list/widgets/players_card.dart [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+                                ),
+                              ),
+                              Text(
+                                player.calculateAge.toString(),
+                                style: const TextStyle(
+                                  color: whiteColor,
+                                ),
+                              ),
+                            ],
+                          ),
```
[the remark this answers is not in the document]
Comment (juliomejia97) [reply lib/src/modules/players_list/widgets/players_card.dart:82]: I am going to leave it as it is. To have work to be done by @leonardomn12 and @BrayanGarcia0

**Pull request merged**